cmd/integration/harness.go

```diff
@@ -86,6 +83,7 @@ func main() {
 		log.With().Error("harness: an error has occurred while generating a new harness:", log.Err(err))
 		log.Panic("error occurred while generating a new harness")
 	}
+	defer h.server.stop()
```

Comment:
this is unfortunately never executed, because `dummyChan` is never closed or written to so main blocks on line 103

Reply:
Refactored the whole main function.